src/server/v2.0/handler/project.go

```diff
@@ -63,3 +241,380 @@ func (a *projectAPI) GetLogs(ctx context.Context, params operation.GetLogsParams
+		params.Project.CVEAllowlist.ProjectID = params.ProjectID
+		lib.JSONCopy(&p.CVEAllowlist, params.Project.CVEAllowlist)
+	}
+
+	if err := a.projectCtl.Update(ctx, p); err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	return operation.NewUpdateProjectOK()
+}
+
+func (a *projectAPI) deletable(ctx context.Context, projectID int64) (*models.ProjectDeletable, error) {
+	proj, err := a.getProject(ctx, projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &models.ProjectDeletable{Deletable: true}
+	if proj.RepoCount > 0 {
+		result.Deletable = false
+		result.Message = "the project contains repositories, can not be deleted"
+	} else if proj.ChartCount > 0 {
+		result.Deletable = false
+		result.Message = "the project contains helm charts, can not be deleted"
+	}
+
+	return result, nil
+}
+
+func (a *projectAPI) getProject(ctx context.Context, projectID int64, options ...project.Option) (*project.Project, error) {
+	p, err := a.projectCtl.Get(ctx, projectID, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := a.populateProperties(ctx, p); err != nil {
+		return nil, err
+	}
+
+	return p, nil
+}
+
+func (a *projectAPI) validateProjectReq(ctx context.Context, req *models.ProjectReq) error {
+	if req.RegistryID != nil {
+		if *req.RegistryID <= 0 {
+			return errors.BadRequestError(fmt.Errorf("%d is invalid value of registry_id, it should be geater than 0", *req.RegistryID))
+		}
+
+		registry, err := replication.RegistryMgr.Get(*req.RegistryID)
+		if err != nil {
+			return fmt.Errorf("failed to get the registry %d: %v", *req.RegistryID, err)
+		}
+		if registry == nil {
+			return errors.NotFoundError(fmt.Errorf("registry %d not found", *req.RegistryID))
+		}
+		permitted := false
+		for _, t := range config.GetPermittedRegistryTypesForProxyCache() {
+			if string(registry.Type) == t {
+				permitted = true
+				break
+			}
+		}
+		if !permitted {
+			return errors.BadRequestError(fmt.Errorf("unsupported registry type %s", string(registry.Type)))
+		}
+	}
+
+	if req.StorageLimit != nil {
+		hardLimits := types.ResourceList{types.ResourceStorage: *req.StorageLimit}
+		if err := quota.Validate(ctx, quota.ProjectReference, hardLimits); err != nil {
+			return errors.BadRequestError(err)
+		}
+	}
+
+	return nil
+}
+
+func (a *projectAPI) populateProperties(ctx context.Context, p *project.Project) error {
+	if secCtx, ok := security.FromContext(ctx); ok {
+		if sc, ok := secCtx.(*local.SecurityContext); ok {
+			roles, err := pro.ListRoles(sc.User(), p.ProjectID)
+			if err != nil {
+				return err
+			}
+			p.RoleList = roles
+			p.Role = highestRole(roles)
+		}
+	}
+
+	total, err := a.repositoryCtl.Count(ctx, q.New(q.KeyWords{"project_id": p.ProjectID}))
+	if err != nil {
+		return err
+	}
+	p.RepoCount = total
+
+	// Populate chart count property
+	if config.WithChartMuseum() {
+		count, err := api.GetChartController().GetCountOfCharts([]string{p.Name})
+		if err != nil {
```

Comment:
Is it possible to ignore this error? Think about, just chart container is somehow failed to launch, but user still can use Harbor.

Just a idea, it's not necessarily to update code.

Reply:
I agree with this.